### PR TITLE
use access token with raw http client

### PR DESCRIPTION
**Pull request by bennyz**
This would make the client usable with setups that have keycloak enabled, and is the standard way to access the API.

Since we cannot rely on the "exp" field return from oVirt, the current approach refreshes the client every 30 minutes (default session timeout) by requesting a new token or when the token is expired (in the event it doesn't use the default Long.MAX_VALUE).

### CODE REVIEW

**Review by ahadas — APPROVED** [review]
as discussed offline - this method is supported on ovirt 4.3 so that's ok from backward compatibility perspective. without it, forklift is not able to access new deployments of ovirt 4.5

cc @jortel

**Comment by bennyz** [comment]
> > as discussed offline - this method is supported on ovirt 4.3 so that's ok in terms of backward compatibility. without this, forklift is not able to access new deployments of ovirt 4.5
> > cc @jortel
> 
> Does the token expire?

Where can I check for expiry and reconnect if needed? Is system(), in:
https://github.com/konveyor/forklift-controller/blob/7c95747942453b50b9a130097a88af4673719e0b/pkg/controller/provider/container/ovirt/client.go#L147-L163
Expected to be called periodically?

File: pkg/controller/provider/container/ovirt/collector.go
Comment (bennyz) [inline pkg/controller/provider/container/ovirt/collector.go:83]: Made it hardcoded to get approval on the direction, if this is accepted it should be configurable in the provider spec

**Review by mansam — APPROVED** [review]
LGTM, but @jortel should also take a look.

**Review by mnecas — APPROVED** [review]
Just tested with 4.5 with keycloak and 4.4 without and works perfectly.

File: pkg/controller/provider/container/ovirt/client.go
Comment (mnecas) [inline pkg/controller/provider/container/ovirt/client.go:125]: If the user does not enter correct auth credentials it will fail with...
```
{"level":"info","ts":1660222302.1400778,"logger":"provider","msg":"Connection test failed.","reason":"strconv.ParseInt: parsing \"\": invalid syntax"}
```
Because the `response.Expiration` and `response.AccessToken` are empty strings.
The `msg` is correct but it would be better to have some more proper `reason`.

File: pkg/controller/provider/container/ovirt/client.go
Comment (bennyz) [reply pkg/controller/provider/container/ovirt/client.go:53]: From what I chcked, yes:
https://github.com/konveyor/forklift-controller/blob/8d67009dee1bfe152213a9df1948f1f59449fe69/pkg/controller/provider/container/ovirt/collector.go

File: pkg/controller/provider/container/ovirt/client.go
Comment (bennyz) [reply pkg/controller/provider/container/ovirt/client.go:142]: added error check above

**Review by yaacov — APPROVED** [review]
works for me :fire:

**Comment by ahadas** [comment]
@mansam hi, this can help us internally, what blocks this from getting in?

File: pkg/controller/provider/container/ovirt/collector.go
Comment (mansam) [inline pkg/controller/provider/container/ovirt/collector.go:83]: Please fix this now rather than leaving a `TODO`. The Provider spec contains a `settings` map for provider-type-specific fields, which could be used for this, with a default of 30 minutes if a value is not set.

**Review by mansam — CHANGES_REQUESTED** [review]
Looks fine other than the TODO. @jortel would you mind taking a look? This is an inventory change.

File: pkg/controller/provider/container/ovirt/client.go
Comment (ahadas) [inline pkg/controller/provider/container/ovirt/client.go:139]: looking at the error Martin mentioned below, if we return that error back I'm afraid it won't be so clear to users - how about returning an error with better description, i.e., saying that we failed to parse the expiration part

File: pkg/controller/provider/container/ovirt/collector.go
Comment (ahadas) [inline pkg/controller/provider/container/ovirt/collector.go:86]:
```suggestion
			log.Error(err, "Couldn't parse http client timeout, falling back to default")
```

File: pkg/controller/provider/container/ovirt/client.go
Comment (ahadas) [inline pkg/controller/provider/container/ovirt/client.go:39]: how about
```suggestion
	accessTokenExpiration    time.Time
```

File: pkg/controller/provider/container/ovirt/client.go
Comment (ahadas) [inline pkg/controller/provider/container/ovirt/client.go:38]: I don't see a need to store both, we can have just a single field for 'client expiration', no?

**Review by ahadas — COMMENTED** [review]
looks good and many of us use it in order to be able to work with latest ovirt - how about moving this PR to github.com/kubev2v/forklift?

**Comment by ahadas** [comment]
let's post to kubev2v/forklift

**Pull request closed without merging**
